influxdb/_dataframe_client.py

```diff
@@ -255,14 +255,16 @@ def _convert_dataframe_to_json(dataframe,
             "h": 1e9 * 3600,
         }.get(time_precision, 1)
 
+        dataframe_zip = zip(dataframe.index,
+                            self._todict_dropna(dataframe[tag_columns]),
+                            self._todict_dropna(dataframe[field_columns]))
+
         points = [
             {'measurement': measurement,
              'tags': dict(list(tag.items()) + list(tags.items())),
              'fields': rec,
              'time': np.int64(ts.value / precision_factor)}
-            for ts, tag, rec in zip(dataframe.index,
-                                    dataframe[tag_columns].to_dict('record'),
-                                    dataframe[field_columns].to_dict('record'))
+            for ts, tag, rec in dataframe_zip
```

Comment:
Got an error when all fields contain NaN values. I changed this to 
`for ts, tag, rec in dataframe_zip if any(rec)`
to skip the points with all empty fields.